src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/BlazorDebug/BlazorDebugConfigurationProvider.ts

```diff
@@ -20,90 +20,65 @@ export class BlazorDebugConfigurationProvider implements vscode.DebugConfigurati
          * only launch the browser.
          */
         if (configuration.request === 'launch') {
-            /**
-             * If the user is debugging a hosted Blazor WebAssembly application,
-             * then the application server needs to be launched via VS Code's debug
-             * workflow to allow debugging on the server-side logic. If the app is
-             * not hosted, then it can be spun as a standalone process to avoid additional overhead.
-             */
-            if (configuration.hosted) {
-                this.launchHostedApp(folder, configuration);
-            } else {
-                this.launchStandaloneApp(folder, configuration);
-            }
+            await this.launchApp(folder, configuration);
         }
 
-        await this.launchBrowser(folder, configuration);
+        this.vscodeType.debug.onDidStartDebugSession(async event => {
+            if (event.name === SERVER_APP_NAME) {
+                await this.launchBrowser(folder, configuration, event);
+            }
+        });
 
         /**
          * If `resolveDebugConfiguration` returns undefined, then the debugger
          * launch is canceled. Here, we opt to manually launch the browser
-         * configruation using `startDebugging` above instead of returning
+         * configuration using `startDebugging` above instead of returning
          * the configuration to avoid a bug where VS Code is unable to resolve
          * the debug adapter for the browser debugger.
          */
         return undefined;
     }
 
-    private launchHostedApp(folder: vscode.WorkspaceFolder | undefined, configuration: vscode.DebugConfiguration) {
-        if (!configuration.program && !configuration.cwd) {
-            const message = `Must provide 'program' and 'cwd' properties in launch configuration for hosted Blazor WebAssembly apps.`;
-            this.vscodeType.window.showErrorMessage(message);
-        }
+    private async launchApp(folder: vscode.WorkspaceFolder | undefined, configuration: vscode.DebugConfiguration) {
+        const program = configuration.hosted ? configuration.program : 'dotnet';
+        const cwd = configuration.cwd || '${workspaceFolder}';
+        const args = configuration.hosted ? [] : ['run'];
 
         const app = {
-            name: HOSTED_APP_NAME,
+            name: SERVER_APP_NAME,
             type: 'coreclr',
             request: 'launch',
-            preLaunchTask: 'build',
-            program: configuration.program,
-            args: [],
-            cwd: configuration.cwd,
+            prelaunchTask: 'build',
+            program,
+            args,
+            cwd,
             env: {
+                ASPNETCORE_ENVIRONMENT: 'Development',

```

Comment:
You're only passing Development now because of the hosted scenario right?

Reply:
Yep. Our hosted template [only setups up the debugger under development mode](https://github.com/dotnet/aspnetcore/blob/b795ac3546eb3e2f47a01a64feb3020794ca33bb/src/ProjectTemplates/Web.ProjectTemplates/content/ComponentsWebAssembly-CSharp/Server/Startup.cs#L99-L106).